stl/inc/ranges

```diff
@@ -1433,6 +1447,152 @@ namespace ranges {
+        }
+
+        template <class _Other>
+            requires _Pipe_cpp23::_Are_composable_range_adaptor_closures<const _Derived&, _Other>
+        constexpr auto operator|(_Other&& _Right) const& noexcept(
+            noexcept(_Pipe_cpp23::_Pipeline{static_cast<const _Derived&>(*this), _STD forward<_Other>(_Right)})) {
+            return _Pipe_cpp23::_Pipeline{static_cast<const _Derived&>(*this), _STD forward<_Other>(_Right)};
+        }
+
+        template <_Pipe_cpp23::_Is_invocable_range_adaptor_closure<_Derived&> _Rng>
+        friend constexpr auto operator|(_Rng&& _Left, _Derived& _Right)
+#ifdef __EDG__ // TRANSITION, VSO-1222776
+            noexcept(noexcept(_STD declval<_Derived&>()(_STD forward<_Rng>(_Left))))
+#else // ^^^ workaround / no workaround vvv
+            noexcept(noexcept(static_cast<_Derived&>(_Right)(_STD forward<_Rng>(_Left))))
+#endif // TRANSITION, VSO-1222776
+        {
+            return static_cast<_Derived&>(_Right)(_STD forward<_Rng>(_Left));
+        }
+
+        template <_Pipe_cpp23::_Is_invocable_range_adaptor_closure<_Derived&&> _Rng>
+        friend constexpr auto operator|(_Rng&& _Left, _Derived&& _Right)
+#ifdef __EDG__ // TRANSITION, VSO-1222776
+            noexcept(noexcept(_STD declval<_Derived>()(_STD forward<_Rng>(_Left))))
+#else // ^^^ workaround / no workaround vvv
+            noexcept(noexcept(static_cast<_Derived&&>(_Right)(_STD forward<_Rng>(_Left))))
+#endif // TRANSITION, VSO-1222776
+        {
+            return static_cast<_Derived&&>(_Right)(_STD forward<_Rng>(_Left));
+        }
+
+        template <_Pipe_cpp23::_Is_invocable_range_adaptor_closure<const _Derived&> _Rng>
+        friend constexpr auto operator|(_Rng&& _Left, const _Derived& _Right)
+#ifdef __EDG__ // TRANSITION, VSO-1222776
+            noexcept(noexcept(_STD declval<const _Derived&>()(_STD forward<_Rng>(_Left))))
+#else // ^^^ workaround / no workaround vvv
+            noexcept(noexcept(static_cast<const _Derived&>(_Right)(_STD forward<_Rng>(_Left))))
+#endif // TRANSITION, VSO-1222776
+        {
+            return static_cast<const _Derived&>(_Right)(_STD forward<_Rng>(_Left));
+        }
+
+        template <_Pipe_cpp23::_Is_invocable_range_adaptor_closure<const _Derived&&> _Rng>
+        friend constexpr auto operator|(_Rng&& _Left, const _Derived&& _Right)
+#ifdef __EDG__ // TRANSITION, VSO-1222776
+            noexcept(noexcept(_STD declval<const _Derived>()(_STD forward<_Rng>(_Left))))
+#else // ^^^ workaround / no workaround vvv
+            noexcept(noexcept(static_cast<const _Derived&&>(_Right)(_STD forward<_Rng>(_Left))))
+#endif // TRANSITION, VSO-1222776
+        {
+            return static_cast<const _Derived&&>(_Right)(_STD forward<_Rng>(_Left));
+        }
+    };
+
+    namespace _Pipe_cpp23 {
+        template <class _ClosureLeft, class _ClosureRight>
+        struct _Pipeline : range_adaptor_closure<_Pipeline<_ClosureLeft, _ClosureRight>> {

```

Comment:
I think the definition of `_Pipeline` can be shared between C++20 and C++23. The C++20 version also deserves the ref-qualifier for `operator()`.

Reply:
Sorry that I didn't see that! 🤦 Thanks for the hint 😃